akka-remote/src/main/scala/akka/remote/artery/tcp/ssl/SslFactory.scala

```diff
@@ -0,0 +1,98 @@
+package akka.remote.artery.tcp.ssl
+
+import java.io.FileNotFoundException
+import java.io.IOException
+import java.security.GeneralSecurityException
+import java.security.SecureRandom
+
+import akka.event.LogMarker
+import akka.event.MarkerLoggingAdapter
+import akka.japi.Util.immutableSeq
+import akka.remote.artery.tcp.SslTransportException
+import com.typesafe.config.Config
+import javax.net.ssl.SSLContext
+import javax.net.ssl.SSLEngine
+
+private[tcp] class SslFactory(config: Config, val sslManagersProvider: SslManagersProvider, prng: SecureRandom)(
+    protected val log: MarkerLoggingAdapter) {
+
+  val SSLProtocol: String = config.getString("protocol")
+
+  val SSLEnabledAlgorithms: Set[String] =
+    immutableSeq(config.getStringList("enabled-algorithms")).toSet
+  val SSLRequireMutualAuthentication: Boolean = config.getBoolean("require-mutual-authentication")
+  val HostnameVerification: Boolean = config.getBoolean("hostname-verification")
+
+  // log hostname verification warning once
+  if (HostnameVerification)
+    log.debug("TLS/SSL hostname verification is enabled.")
+  else
+    log.info(
+      LogMarker.Security,
+      "TLS/SSL hostname verification is disabled. See Akka reference documentation for more information.")
```

Comment:
Given a `SslManagersProvider` (produces TrustManagers and KeyManagers), a `SecureRandom` and a `Config` this will create `SSLContext`, `SSLEngine`, ...

Cache the whole `SSLFactory` if you need to.